launcher/Application.cpp

```diff
@@ -673,8 +674,36 @@ Application::Application(int &argc, char **argv) : QApplication(argc, argv)
 
         m_settings->registerSetting("UpdateDialogGeometry", "");
 
-        // pastebin URL
-        m_settings->registerSetting("PastebinURL", "https://0x0.st");
+        // This code feels so stupid is there a less stupid way of doing this?
+        {
+            m_settings->registerSetting("PastebinURL", "");
+            QString pastebinURL = m_settings->get("PastebinURL").toString();
+
+            // If PastebinURL hasn't been set before then use the new default: mclo.gs
+            if (pastebinURL == "") {
+                m_settings->registerSetting("PastebinType", PasteUpload::PasteType::Mclogs);
+                m_settings->registerSetting("PastebinCustomAPIBase", "");
+            }
+            // Otherwise: use 0x0.st
+            else {
+                // The default custom endpoint would usually be "" (meaning there is no custom endpoint specified)
+                // But if the user had customised the paste URL then that should be carried over into the custom endpoint.
+                QString defaultCustomEndpoint = (pastebinURL == "https://0x0.st") ? "" : pastebinURL;
```

Comment:
We could switch users who have not changed the URL to mclo.gs as well

Reply:
?

Reply:
I don't know if it's good to switch people who selected 0x0.st because they _want_ to use 0x0.st over to mclo.gs completely automatically.

Reply:
people who actually want 0x0.st could still change it manually back

Reply:
> people who actually want 0x0.st could still change it manually back

Yeah, if I will do that I need to add some sort of indication to people that the default paste service was changed and an option to change it back to 0x0.st

Reply:
Done